assembly/src/assembler/instruction/field_ops.rs

```diff
@@ -95,22 +95,89 @@ pub fn pow2(span: &mut SpanBuilder) -> Result<Option<CodeBlock>, AssemblyError>
     Ok(None)
 }
 
+/// Computes `ilog2` for the number at the top of the stack, and its corresponding power-of-two.
+///
+/// Input: [n, ...]
+/// Output: [pow2, ilog2, ...]
+/// Cycles: 52
+pub fn ilog2(span: &mut SpanBuilder) -> Result<Option<CodeBlock>, AssemblyError> {
+    // validate `n` is not `0`
+    span.push_ops([Dup0, Eqz, Not, Assert]);
+
+    // push the result of the ilog2 operation to the advice stack
+    span.add_decorator(Decorator::Advice(AdviceInjector::ILog2))?;
```

Comment:
This is different than some of the existing instruction, like `adv.u64div`, that use the advice provider only to push the data to the advice stack. I went with this approach because I couldn't think of a use case were the user would want the value that is not validated, and having a instruction is nicer to use than a procedure.